RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -0,0 +1,586 @@
+
+      seeds.swap(cleanSeedTracks);
+
+#ifdef DEBUG
+      {
+        const int ns2 = seeds.size();
+        printf("Number of CMS seeds before %d --> after %d cleaning\n", ns, ns2);
+
+        for (int it = 0; it < ns2; it++) {
+          const Track &ss = seeds[it];
+          printf("  %3i q=%+i pT=%7.3f eta=% 7.3f nHits=%i label=% i\n",
+                 it,
+                 ss.charge(),
+                 ss.pT(),
+                 ss.momEta(),
+                 ss.nFoundHits(),
+                 ss.label());
+        }
+      }
+#endif
+
+#ifdef DEBUG
+      std::cout << "AFTER seed cleaning " << seeds.size() << std::endl;
+#endif
+
+      return seeds.size();
+    }
+
+    //=========================================================================
+    // Duplicate cleaning
+    //=========================================================================
+
+    void find_duplicates(TrackVec &tracks) {
+      const auto ntracks = tracks.size();
+      float eta1, phi1, pt1, deta, dphi, dr2;
+      //float ch1;
+      if (ntracks == 0) {
+        return;
+      }
+      for (auto itrack = 0U; itrack < ntracks - 1; itrack++) {
+        auto &track = tracks[itrack];
+        if (track.algoint() == 9)
+          continue;
+        if (track.algoint() == 10)
+          continue;
+        eta1 = track.momEta();
+        phi1 = track.momPhi();
+        //ch1  = track.charge();
+        pt1 = track.pT();
+        for (auto jtrack = itrack + 1; jtrack < ntracks; jtrack++) {
+          auto &track2 = tracks[jtrack];
+          if (track.label() == track2.label())
+            continue;
+          if (track.algoint() != track2.algoint())
+            continue;
+
+          //if (ch1 != track2.charge()) continue;
+
+          deta = std::abs(track2.momEta() - eta1);
+          if (deta > Config::maxdEta)
+            continue;
+
+          dphi = std::abs(squashPhiMinimal(phi1 - track2.momPhi()));
+          if (dphi > Config::maxdPhi)
+            continue;
+
+          float maxdR = Config::maxdR;  // maxdR = 0.0025
+          float maxdRSquared = maxdR * maxdR;
+          if (std::abs(eta1) > 2.5f)
+            maxdRSquared *= 16.0f;
+          else if (std::abs(eta1) > 1.44f)
+            maxdRSquared *= 9.0f;
+          dr2 = dphi * dphi + deta * deta;
+          if (dr2 < maxdRSquared) {
+            //Keep track with best score
+            if (track.score() > track2.score())
+              track2.setDuplicateValue(true);
+            else
+              track.setDuplicateValue(true);
+            continue;
+          } else {
+            if (pt1 == 0)
+              continue;
+            if (track2.pT() == 0)
+              continue;
+
+            if (std::abs((1 / track2.pT()) - (1 / pt1)) < Config::maxdPt) {
+              if (Config::useHitsForDuplicates) {
+                float numHitsShared = 0;
+                for (int ihit2 = 0; ihit2 < track2.nTotalHits(); ihit2++) {
+                  int hitidx2 = track2.getHitIdx(ihit2);
```

Comment:
const here and below

Reply:
Addressed in Round 1 follow-up PR.
trackreco/cmssw#75 https://github.com/trackreco/cmssw/pull/75/commits/e9834b4a33dab0a34078a4861b825f0e1ebdb64d